### PR TITLE
SDXL improvements (and support for Draft+)   [DRAFT PR]

### PR DESCRIPTION
**Pull request by rohitrango**
# What does this PR do ?

This PR adds additional functionality to SDXL class of models to support Draft+, and:
* bug fix for adapter control in LoRA (LinearParallelAdapter)
* conversion script for SDXL from huggingface weights to nemo (existing converter script doesnt work)
* additional inference config file to load converted hf weights
* added FSDP strategy for multimodal models
* function to return adapter config, similar to function that returns the adapter itself

Code definitely needs cleanup.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/multimodal/models/text_to_image/stable_diffusion/ldm/autoencoder.py [hunk 1]
```diff
@@ -316,6 +317,7 @@ def __init__(
         ignore_keys=[],
         image_key="image",
         colorize_nlabels=None,
+        from_NeMo=False,
```
Comment (yaoyu-33) [inline nemo/collections/multimodal/models/text_to_image/stable_diffusion/ldm/autoencoder.py:320]: use `from_nemo` lowercase

File: nemo/collections/multimodal/modules/stable_diffusion/diffusionmodules/openaimodel.py [hunk 2]
```diff
@@ -1205,7 +1217,12 @@ def _state_key_mapping(self, state_dict: dict):
     def _load_pretrained_model(self, state_dict, ignore_mismatched_sizes=False, from_NeMo=False):
         state_dict = self._strip_unet_key_prefix(state_dict)
         if not from_NeMo:
+            print("creating state key mapping from HF")
```
Comment (yaoyu-33) [inline nemo/collections/multimodal/modules/stable_diffusion/diffusionmodules/openaimodel.py:1220]: remove all debug prints

File: nemo/collections/multimodal/parts/utils.py [hunk 3]
```diff
@@ -321,7 +334,9 @@ def setup_trainer_and_model_for_inference(
         )
 
     else:
-        raise ValueError(f"Unrecognized checkpoint type: {cfg.model.restore_from_path}")
+        model = model_provider(cfg=cfg.model, trainer=trainer)
+        # raise ValueError(f"Unrecognized checkpoint type: {cfg.model.restore_from_path}")
```
Comment (yaoyu-33) [inline nemo/collections/multimodal/parts/utils.py:338]: clean

File: nemo/collections/multimodal/models/text_to_image/stable_diffusion/ldm/autoencoder.py (hunk 1 above)
Comment (rohitrango) [reply nemo/collections/multimodal/models/text_to_image/stable_diffusion/ldm/autoencoder.py:320]: I've tried to stay consistent with the format in `nemo/collections/multimodal/modules/stable_diffusion/diffusionmodules/openaimodel.py` (L1220).

File: nemo/collections/multimodal/modules/stable_diffusion/diffusionmodules/openaimodel.py (hunk 2 above)
Comment (rohitrango) [reply nemo/collections/multimodal/modules/stable_diffusion/diffusionmodules/openaimodel.py:1220]: done

File: nemo/collections/multimodal/parts/utils.py (hunk 3 above)
Comment (rohitrango) [reply nemo/collections/multimodal/parts/utils.py:338]: clean as in clean the comment or undo the change?

**Review by yaoyu-33 — APPROVED** [review]
